File: mediator/typescript/chat-user.ts
import {User} from "./user";
import {IChatRoomMediator} from "./chat-room.interface";

// Colleague (User) soyutundan türer.
// Mediator sayesinde diğer sınıf örnekler ile iletişim kurar.
// UML diyagramındaki ConcreteColleague yapısına denk gelmektedir.
export class ChatUser extends User {
  constructor(public chatRoomMediator: IChatRoomMediator) {
    super(chatRoomMediator);
  }
}

File: mediator/go/chat-user.go
package main

// Colleague (User) soyutundan türer.
// Mediator sayesinde diğer sınıf örnekler ile iletişim kurar.
// UML diyagramındaki ConcreteColleague yapısına denk gelmektedir.
type chatUser struct {
	User *User
}

func NewChatUser(mediator ChatRoom, id int, name string) *chatUser {
	return &chatUser{
		User: NewUser(mediator, id, name),
	}
}
